src/components/implementation/no_interface/scheddev/sl_lock.c

```diff
@@ -0,0 +1,67 @@
+#include "sl.h"
+#include "sl_lock.h"
+
+void
+sl_lock_init(struct sl_lock *lock)
+{
+    assert(lock);
+    *lock = SL_LOCK_STATIC_INIT();
+}
+
+thdid_t
+sl_lock_holder(struct sl_lock *lock)
+{
+    assert(lock);
```

Comment:
I'm not sold on all of these `assert(lock)`s.  They only check if the client passes in `NULL`, which is a bug that would be near trivial to diagnose with `objdump`.  They have a very real cost (binary size and branch target buffer capacity), and don't provide a real benefit.